### PR TITLE
Add dpnp.rollaxis

### PR DESCRIPTION
**Pull request by densmirn**

### CODE REVIEW

File: dpnp/dpnp_iface_manipulation.py [hunk 1]
```diff
@@ -135,6 +137,53 @@ def repeat(x1, repeats, axis=None):
...
+        Output array.
+
+    See Also
+    --------
+    moveaxis, roll
+
+    """
+
+    if not use_origin_backend(a):
+        def use_dpnp_backend(a, axis, start):
```
Comment (shssf) [inline dpnp/dpnp_iface_manipulation.py:166]: Could you please avoid creating nested functions?

File: dpnp/dpnp_utils.pyx [hunk 2]
```diff
@@ -216,3 +217,20 @@ cpdef long _get_linear_index(key, tuple shape, int ndim):
     else:
         li = key
     return li
+
+
+def dp2ndarray(arr):
```
Comment (shssf) [inline dpnp/dpnp_utils.pyx:222]:
```suggestion
def dp2nd_array(arr):
```

File: dpnp/dpnp_utils.pyx [hunk 3]
```diff
@@ -216,3 +217,20 @@ cpdef long _get_linear_index(key, tuple shape, int ndim):
...
+    """Convert dparray to ndarray"""
+    return dpnp.asnumpy(arr) if isinstance(arr, dparray) else arr
+
+
+def nd2dparray(arr):
```
Comment (shssf) [inline dpnp/dpnp_utils.pyx:227]:
```suggestion
def nd2dp_array(arr):
```

File: dpnp/dpnp_iface_manipulation.py [hunk 4]
```diff
@@ -135,6 +137,53 @@ def repeat(x1, repeats, axis=None):
...
+            if start < -a.ndim or start > a.ndim:
+                return False
+
+            return True
+
+        if use_dpnp_backend(a, axis, start):
+            start_norm = start + a.ndim if start < 0 else start
+            destination = start_norm - 1 if start_norm > axis else start_norm
+
+            return moveaxis(a, axis, destination)
```
Comment (shssf) [inline dpnp/dpnp_iface_manipulation.py:180]:
```suggestion
            return dpnp.moveaxis(a, axis, destination)
```

**Comment by shssf** [comment]
@densmirn One more idea. Perhaps it is better to move dp2nd and nd2dp function to dparray.pyx. 
these functions logically works with dparray and its representation. 
What do you think?

**Comment by densmirn** [comment]
> @densmirn One more idea. Perhaps it is better to move dp2nd and nd2dp function to dparray.pyx.
> these functions logically works with dparray and its representation.
> What do you think?

Do you mean something like this?
```python
cdef class dparray:
    # ...
    @classmethod
    def from_ndarray(cls, arr):
        """Convert ndarray to dparray"""
        if not isinstance(arr, numpy.ndarray):
            return arr

        result = cls(arr.shape, dtype=arr.dtype)
        for i in range(result.size):
            result._setitem_scalar(i, arr.item(i))

        return result

    @staticmethod
    def to_ndarray(arr):
        """Convert dparray to ndarray"""
        return dpnp.asnumpy(arr) if isinstance(arr, dparray) else arr

def rollaxis(a, axis, start=0):
    # ...
    result = numpy.rollaxis(dparray.to_ndarray(a), axis, start)

    return dparray.from_ndarray(result)
```

**Comment by shssf** [comment]
@densmirn yes but with a bit smaller code
cdef class dparray:
    # ...
    @classmethod
    def from_ndarray(cls, arr):
        """Convert ndarray to dparray"""
        if not isinstance(arr, numpy.ndarray):
            return arr

        result = cls(arr.shape, dtype=arr.dtype)
        for i in range(result.size):
            result._setitem_scalar(i, arr.item(i))

        return result

    def dp2nd_array(self):
        """Convert dparray to ndarray"""
        return asnumpy(arr) if isinstance(arr, dparray) else arr

**Pull request merged**